core/src/data/key.rs

```diff
@@ -130,12 +132,31 @@ impl TryFrom<Value> for Key {
             Uuid(v) => Ok(Key::Uuid(v)),
             Null => Ok(Key::None),
             Map(_) => Err(KeyError::MapTypeKeyNotSupported.into()),
-            List(_) => Err(KeyError::ListTypeKeyNotSupported.into()),
+            List(values) => Key::try_from(values),
             Point(_) => Err(KeyError::PointTypeKeyNotSupported.into()),
         }
     }
 }
 
+impl TryFrom<Vec<Value>> for Key {
+    type Error = Error;
+
+    fn try_from(values: Vec<Value>) -> Result<Self> {
+        Ok(Key::List(
+            values
+                .into_iter()
+                .map(Key::try_from)
+                .collect::<Result<Vec<_>>>()?,
+        ))
+    }
+}
+
+impl From<Vec<Key>> for Key {
+    fn from(keys: Vec<Key>) -> Self {
+        Key::List(keys)
+    }
+}
+
```

Comment:
nit. we might not need this.
only existing use case is in test-suite primary_key case but we can simply wrap `Key::List` rather than using from impl.

Reply:
Done